examples/simple-chatbot/server/bot-openai.py

```diff
@@ -69,6 +81,68 @@
 quiet_frame = sprites[0]  # Static frame for when bot is listening
 talking_frame = SpriteFrame(images=sprites)  # Animation sequence for when bot is talking
 
+#
+# RTVI events for Pipecat client UI
+#
+# rtvi = None
+
+
+class CustomProcessor(FrameProcessor):
+    """Processes weather-related function calls.
+
+    This processor handles the function call to fetch weather data and
+    manages the response.
+    """
+
+    # currently does nothing but tracks waiting calls but could be used
+    waiting_calls: Dict[str, FunctionCallParams] = {}
+
+    def __init__(self):
+        super().__init__()
+
+    async def fetch_weather(self, params: FunctionCallParams):
+        print("Fetching weather data...", params)
+        await params.llm.push_frame(TTSSpeakFrame("Let me check on that."))
+        await rtvi.handle_function_call(params)
+        self.waiting_calls[params.tool_call_id] = params
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        """Process incoming frames and handle function calls.
+
+        Args:
+            frame: The incoming frame to process
+            direction: The direction of frame flow in the pipeline
+        """
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, FunctionCallResultFrame):
+            print("Function call result:", frame.tool_call_id, frame.result)
+            if "weather" in frame.result and "condition" in frame.result["weather"]:
+                frame.result["weather"]["condition"] = "hazy"
+            if frame.tool_call_id in self.waiting_calls:
+                del self.waiting_calls[frame.tool_call_id]
+        # elif isinstance(frame, RTVIClientMessageFrame):
```

Comment:
Maybe a comment noting why you might choose this pattern over the `@rtvi.event_handler("on_client_message")` pattern (if there's a reason to do it this way).

I think I understand why this code is here and commented-out, though: you want to be able to quickly uncomment it and test it with this alternate pattern.

Reply:
i'll chat with @markbackman and see if he has a good idea for better showing this stuff off and if this is the right example to do so in.